src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AnnotationFieldDefHandler.java

```diff
@@ -0,0 +1,100 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.indentation;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Handler for annotation field definitions.
+ * Annotation field defintion closely resemble the method defintions and therefore extends
+ * MethodDefHandler.
+ *
+ * @author Vikramaditya Kukreja
+ */
+public class AnnotationFieldDefHandler extends MethodDefHandler {
+    /**
+     * Construct an instance of this handler with the given indentation check,
+     * abstract syntax tree, and parent handler.
+     *
+     * @param indentCheck   the indentation check
+     * @param ast           the abstract syntax tree
+     * @param parent        the parent handler
+     */
+    public AnnotationFieldDefHandler(IndentationCheck indentCheck,
+                                     DetailAST ast, AbstractExpressionHandler parent) {
+        super(indentCheck, ast, parent);
+    }
+
+    @Override
```

Comment:
Also for the 1st example, where it is assumed I am working with only MethodDefHandler,

take these 2 cases - 
Case 1 :
```java
@ //indent:0 exp:0
 interface //indent:1 exp:0 warn
MyAnnotationWrapped { //indent:0 exp:0
    int value() default -1; //indent:4 exp:4
    AnnotationWithTarget //indent:4 exp:4
        .AnnotationInnerLineWrap field(); //indent:8 exp:8 //LINE IN QUESTION
} //indent:0 exp:0
```
When CS runs on this, i get error message that it `exp` should be 4.


2. Case 2 : I try to do as suggested
```java
@ //indent:0 exp:0
 interface //indent:1 exp:0 warn
MyAnnotationWrapped { //indent:0 exp:0
    int value() default -1; //indent:4 exp:4
    AnnotationWithTarget //indent:4 exp:4
    .AnnotationInnerLineWrap field(); //indent:4 exp:4 //LINE IN QUESTION
} //indent:0 exp:0
```
Here I get the error that `exp` should be 8